src/core/modules/AtmosphereShowMySky.cpp

```diff
@@ -517,6 +501,13 @@ void AtmosphereShowMySky::regenerateGrid()
 //	ppxmax = ppxatmo = conf->value("landscape/ppxatmo", 1).toInt();
 	gridMaxY = conf->value("landscape/atmosphereybin", 44).toInt();
 	gridMaxX = std::floor(0.5+gridMaxY*(0.5*std::sqrt(3.0))*width/height);
+	const auto rppx=qRound(sqrt(ppxatmo));
```

Comment:
Although this indeed does improve the looks as seen in your screenshot, I'm still not convinced it fixes the problem at its origin. Why did the grid even change before this patch?

Reply:
It seems that the size of the grid stays the same, but we have a lot fewer 
- 1/ppxatmo^2 (without the patch)
- 1/ppxatmo (with the patch)

elements due to the smaller render target.

The final display is then simply scaled up with ppxatmo.

Try this representation of triangles for fun...


```diff
diff --git a/src/core/modules/AtmosphereShowMySky.cpp b/src/core/modules/AtmosphereShowMySky.cpp
index 4f74c1e..508bcce 100644
--- a/src/core/modules/AtmosphereShowMySky.cpp
+++ b/src/core/modules/AtmosphereShowMySky.cpp
@@ -446,7 +446,7 @@
 				std::size_t shift=0;
 				for (int y=0;y<gridMaxY;++y)
 				{
-					GL(gl.glDrawElements(GL_TRIANGLE_STRIP, (gridMaxX+1)*2, GL_UNSIGNED_SHORT,
+					GL(gl.glDrawElements(GL_TRIANGLES, (gridMaxX+1)*2, GL_UNSIGNED_SHORT,
 										 reinterpret_cast<void*>(shift)));
 					shift += (gridMaxX+1)*2*2;
 				}
```

Reply:
Aha, so that's the problem. I suspected it, but misled myself during thinking :) .

The grid is generated assuming `width` and `height` representing the full-resolution viewport. What your change should be is  as follows:
```diff
@@ -511,7 +511,8 @@ AtmosphereShowMySky::~AtmosphereShowMySky()
 
 void AtmosphereShowMySky::regenerateGrid()
 {
-    const float width=viewport[2], height=viewport[3];
+    const float width = viewport[2]/ppxatmo, height = viewport[3]/ppxatmo;
     QSettings* conf = StelApp::getInstance().getSettings();
 //  flagDynamicResolution = conf->value("landscape/flag_dynamic_resolution", false).toBool();
 //  ppxmax = ppxatmo = conf->value("landscape/ppxatmo", 1).toInt();
```
Just be sure to regeneration the grid on each switch between full and reduced resolutions.

Reply:
I just changed it - looks good!